client/ayon_deadline/addon.py

```diff
@@ -38,6 +39,11 @@ def initialize(self, studio_settings):
 
         self.deadline_servers_info = deadline_servers_info
 
+        self._pools_by_server_name = {}
```

Comment:
We should probably invalidate the cache after some time? There is helper `NestedCacheItem` class for that

```
def _initialization(self):
    ...
    self._limit_groups_by_server_name = NestedCacheItem(lifetime=5*60)
    ...

def limit_groups_by_server_name(self, server_name: str) -> List[str]:
        limit_groups_cache = self._limit_groups_by_server_name[server_name]
        if not limit_groups_cache.is_valid:
            dl_server_info = self.deadline_servers_info.get(server_name)

            auth = (dl_server_info["default_username"],
                    dl_server_info["default_password"])
            limit_groups = self.get_deadline_limit_groups(
                dl_server_info["value"],
                auth
            )
            limit_groups_cache.update_data(limit_groups)

        return limit_groups_cache.get_data()

Reply:
We know about artists who have opened Nuke for a week. We should probably invalidate, cached values after some time.